app/models/edition.rb

```diff
@@ -465,20 +465,22 @@ def included_in_statistics_feed?
   # @!endgroup
 
   def create_draft(user)
-    unless published?
-      raise "Cannot create new edition based on edition in the #{state} state"
-    end
-
-    ignorable_attribute_keys = %w(id type state created_at updated_at change_note
-                                  minor_change force_published scheduled_publication)
-    draft_attributes = attributes.except(*ignorable_attribute_keys)
-      .merge("state" => "draft", "creator" => user, "previously_published" => previously_published)
+    ActiveRecord::Base.transaction do
+      unless published?
+        raise "Cannot create new edition based on edition in the #{state} state"
+      end
```

Comment:
You could do this prior to the transaction, as it's not modifying any state.

That would also avoid beginning the transaction, then rolling back if the state is wrong.